src/org/labkey/test/pages/assay/ManageAssayQCStatesPage.java

```diff
@@ -0,0 +1,99 @@
+package org.labkey.test.pages.assay;
+
+import org.labkey.test.Locator;
+import org.labkey.test.WebDriverWrapper;
+import org.labkey.test.WebTestHelper;
+import org.labkey.test.pages.LabKeyPage;
+import org.labkey.test.pages.study.QCStateTableRow;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ManageAssayQCStatesPage extends LabKeyPage<ManageAssayQCStatesPage.ElementCache>
```

Comment:
This looks to have a lot of duplicate code with `ManageDatasetQCStatesPage`. Could a bunch of this be pulled into a shared base class?

Reply:
Potentially, yes.
But the underlying feature is still in flux.  I didn't want to commit to a single base class when the product implementations are separate and could possibly change independent of each other.  I've been under some time pressure to get this done, made this call based on that concern.